tests/templates/test_layers/test_cv_neural_net.py

```diff
@@ -0,0 +1,306 @@
+            assert gate.wires.labels == tuple(expected_wires[i])
+
+    def test_custom_wire_labels(self, tol):
+        """Test that template can deal with non-numeric, nonconsecutive wire labels."""
+        shapes = expected_shapes(1, 3)
+        weights = [np.random.random(shape) for shape in shapes]
+
+        dev = DummyDevice(wires=3)
+        dev2 = DummyDevice(wires=["z", "a", "k"])
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.templates.CVNeuralNetLayers(*weights, wires=range(3))
+            return qml.expval(qml.Identity(0))
+
+        @qml.qnode(dev2)
+        def circuit2():
+            qml.templates.CVNeuralNetLayers(*weights, wires=["z", "a", "k"])
+            return qml.expval(qml.Identity("z"))
+
+        circuit()
+        circuit2()
+
+        assert np.allclose(dev._state[0], dev2._state[0], atol=tol, rtol=0)
+        assert np.allclose(dev._state[1], dev2._state[1], atol=tol, rtol=0)
+
+
+class TestInputs:
+    """Test inputs and pre-processing."""
+
+    def test_cvqnn_layers_exception_nlayers(self):
+        """Check exception if inconsistent number of layers"""
+        shapes = expected_shapes(1, 2)
+        weights = [np.random.random(shape) for shape in shapes[:-1]]
+        weights += [np.random.random((2, shapes[-1][1]))]
+
+        dev = DummyDevice(wires=2)
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.templates.CVNeuralNetLayers(*weights, wires=range(2))
+            return qml.expval(qml.X(0))
+
+        with pytest.raises(ValueError, match="The first dimension of all parameters"):
+            circuit()
+
+    def test_cvqnn_layers_exception_second_dim(self):
+        """Check exception if weong dimension of weights"""
+        shapes = expected_shapes(1, 2)
+        weights = [np.random.random(shape) for shape in shapes[:-1]]
+        weights += [np.random.random((1, shapes[-1][1] - 1))]
+
+        dev = DummyDevice(wires=2)
+
+        @qml.qnode(dev)
+        def circuit():
+            qml.templates.CVNeuralNetLayers(*weights, wires=range(2))
+            return qml.expval(qml.X(0))
+
+        with pytest.raises(ValueError, match="Got unexpected shape for one or more parameters"):
+            circuit()
+
+
+class TestAttributes:
+    """Test methods and attributes."""
+
+    @pytest.mark.parametrize(
+        "n_layers, n_wires",
+        [
+            (2, 3),
+            (2, 1),
+            (2, 2),
+        ],
+    )
+    def test_shapes(self, n_layers, n_wires, tol):
+        """Test that the shape method returns the correct shapes for
+        the weight tensors"""
+
+        shapes = qml.templates.CVNeuralNetLayers.shape(n_layers, n_wires)
+        expected = expected_shapes(n_layers, n_wires)
+
+        assert np.allclose(shapes, expected, atol=tol, rtol=0)
+
+
+def circuit_template(*weights):
+    qml.templates.CVNeuralNetLayers(*weights, range(2))
+    return qml.expval(qml.X(0))
+
+
+# def circuit_decomposed(*weights):
```

Comment:
This has to be uncommented once the interferometer issue is solved

Reply:
Reminder to uncomment!

Reply:
Ah, thanks, I forgot!